app/models/order.rb

```diff
@@ -0,0 +1,48 @@
+class Order < ActiveRecord::Base
+  has_many :order_items
+  has_many :products, :through => :order_items
+  validate :must_have_order_items
+  validates :customer_email, presence: true, on: :update, if: :paid?
+  validates :street_address, presence: true, on: :update, if: :paid?
+  validates :city, presence: true, on: :update, if: :paid?
+  validates :zip_code, presence: true, length: {minimum: 5 }, on: :update, if: :paid?
+  validates :name_on_card, presence: true, on: :update, if: :paid?
+  validates :card_number, presence: true, on: :update, if: :paid?
+  validates :customer_card_exp_month, presence: true, numericality: { greater_than: 0 }, on: :update, if: :paid?
+  validates :customer_card_exp_year, presence: true, numericality: { greater_than: 0 }, on: :update, if: :paid?
+  validates :billing_zip_code, presence: true, length: {minimum: 5 }, on: :update, if: :paid?
+  validates :security_code, presence: true, length: {minimum: 3 }, on: :update, if: :paid?
+  #validate :expiration_date_cannot_be_in_the_past
+
+  #this validation ruined all our specs and we can't tell why. Leaving this in to ask Charles why it had that effect
```

Comment:
It would be better if you guys deleted this block of commented-out-code and ask your question through a comment on your PR instead